### ISSUE TITLE
Comply with cookie law

### ISSUE THREAD

**Issue by xavez** [iRail/hyperRail#220]
In order to comply with cookie law in Belgium, I think we have to put in one of those notices. We can only set cookies after the user agrees. 

Another option is to remove Google Analytics. Which would be my preference because it increases burden on the user and it is a privacy liability for users of iRail.

**Comment by pietercolpaert** [comment]
We have a cookie to keep the language choice and we indeed have google analytics.

The latter was chosen on the 15th of September 2008 by @tuinslak, which was a good decision at that time. We have the google analytics since then and scrolling through 8 years of iRail.be is pretty cool. Changing to another system would be possible, yet would come with the migration work I'm not willing to do at this moment (other issues take more priority).

An idea to comply to the law would be to add a disclaimer on the language choice page, as when no cookie has been set yet, this page will turn up?

**Comment by xavez** [comment]
Makes sense @pietercolpaert. If you can tell me what it has to say, legally speaking, I can implement it.

**Comment by Haroenv** [comment]
This is the technical document: http://ec.europa.eu/ipg/basics/legal/cookies/index_en.htm

**Comment by Tjoosten** [comment]
https://github.com/spatie/laravel-cookie-consent here an awesome package for it.

**Comment by Tjoosten** [comment]
I'm trying something on this.

**Comment by Tjoosten** [comment]
@Haroenv can u tell me where the base layout for the app is located. The only thing i still need to do is add a `@include`.

**Comment by Haroenv** [comment]
There's no *one* main layout, multiple layouts for multiple pages. Maybe that should be changed so it all inherits some base stuff? https://github.com/iRail/hyperRail/blob/master/resources/views/route/planner.blade.php is one for example, as is https://github.com/iRail/hyperRail/blob/master/resources/views/language.blade.php and https://github.com/iRail/hyperRail/blob/529bdd22f4231c1ab41ab5bc2389bfd7d612d731/resources/views/contributors/home.blade.php and some others (search https://github.com/iRail/hyperRail/search?utf8=✓&q=doctype)

**Comment by Tjoosten** [comment]
Will look after it.

**Comment by Bertware** [comment]
@pietercolpaert noticed this issue now, was this fixed? Otherwise the google analytics part can just be deleted, and functional language selection cookies don't need approval or popups as far as I know.

**Comment by pietercolpaert** [comment]
Ok to remove ganalytics
